src/Tizen.Core/Tizen.Core/Task.cs

```diff
@@ -21,81 +21,90 @@
+        /// <param name="id">The ID of the task.</param>
+        /// <exception cref="ArgumentException">Thrown when the id argument is invalid or already exists.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when out of memory.</exception>
+        /// <remarks>
+        /// The constructor throws an exception when the id already exists.
+        /// By default, the task creates a thread. However, if the id is main, it works without creating a thread. 
+        /// The 'main' task will be operated in the main thread.
+        /// </remarks>
         /// <example>
         /// <code>
         /// 
-        /// var coreTask = new TCoreTask("Worker");
-        /// coreTask.Run();
+        /// TizenCore.Initialize();
+        /// var task = new Task("Worker");
+        /// task.Run();
         /// 
         /// </code>
         /// </example>
         /// <since_tizen> 12 </since_tizen>
-        public TCoreTask(string id)
+        public Task(string id)
         {
             bool useThread = (id == "main") ? false : true;
             Interop.LibTizenCore.ErrorCode error = Interop.LibTizenCore.TizenCore.TaskCreate(id, useThread, out  _handle);
             TCoreErrorFactory.CheckAndThrownException(error, "Failed to create task");
             _coreTaskMap[id] = this;
         }
 
-        private TCoreTask(IntPtr handle)
+        internal Task(IntPtr handle)
         {
             _handle = handle;
             _disposed = true;
         }
 
         /// <summary>
-        /// Finalizes the TCoreTask class.
+        /// Finalizes the Task class.
         /// </summary>
         /// <since_tizen> 12 </since_tizen>
-        ~TCoreTask()
+        ~Task()
         {
              Dispose(false);
         }
 
+
         /// <summary>
-        /// Dispatches an asynchronous message to a main loop of the TCoreTask.
+        /// Posts an action to be executed later.
         /// </summary>
-        /// <remarks>
-        /// The asynchronous message will be delivered to the main loop of the TCoreTask.
-        /// </remarks>
-        /// <param name="action">The action callback.</param>
-        /// <exception cref="ArgumentNullException">Thrown when the argument is null.</exception>
+        /// <param name="action">The action callback to post.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the action argument is null.</exception>
         /// <exception cref="ArgumentException">Thrown when failed because of the instance is invalid.</exception>
         /// <exception cref="OutOfMemoryException">Thrown when out of memory.</exception>
+        /// <remarks>
+        /// The action will be stored in the internal map using its unique identifier.
+        /// Then it will be added as an idle job to the main loop of the task..
+        /// The action callback will be executed by the main loop of the task.
+        /// If there was any error during this process, an appropriate exception will be thrown.
```

Comment:
Right, I forgot about that, how to catch exceptions in a posted Action/function? or maybe they cannot be caught but only processed in something like AppDomain.UnhandledException event? Do they crash/terminate application? Such information would be nice to describe here. Not necessary at this moment, it could be added later.

Reply:
Done